adlfs/tests/conftest.py

```diff
@@ -37,7 +37,8 @@ def storage(host):
     conn_str = f"DefaultEndpointsProtocol=http;AccountName={ACCOUNT_NAME};AccountKey={KEY};BlobEndpoint={URL}/{ACCOUNT_NAME};"  # NOQA
 
     bbs = BlobServiceClient.from_connection_string(conn_str=conn_str)
-    bbs.create_container("data")
+    if "data" not in [c["name"] for c in bbs.list_containers()]:
```

Comment:
I was running into some race condition locally where the `data` container would already be created (maybe by another test), and the `create_container` call would throw if the container already existed by the time this test ran. If Azurite was a little more consistent, I wouldn't have to do this.

Reply:
Hm, it's fishy! This fixture is supposed to remove the container at the end of every test. Are there any tests that use a different fixture?
If opting not to recreate, we don't really know if the container contents are as we expect.

Reply:
That does surprise me too :/

Unfortunately, we can't (easily) include a random hash in the container name since it shows up in all the paths, and the tests are hardcoded to `data`.

I'm OK with this for now, but let's keep an eye on weird test failures in the future...